src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/FilingsService.java

```diff
@@ -124,6 +127,16 @@ private void setFilingApiData(
       publicDataRetrievalService.initialisePublicData(
           submissionDto.getEntityNumber(), passThroughTokenHeader);
       privateDataRetrievalService.initialisePrivateData(submissionDto.getEntityNumber());
+
+      var publicPrivateDataCombiner = new PublicPrivateDataCombiner(
```

Comment:
In future I would like to see Spring components and auto-wiring being used.

Reply:
You're saying in the future so you're ok with it how it is now?

Only Swetha and others are waiting for this and I really don't think another substantial change is wise

Reply:
It's OK for now.